sympy/core/numbers.py

```diff
@@ -1420,8 +1420,19 @@ def __add__(self, other):
             if isinstance(other, Integer):
                 return Rational(self.p + self.q*other.p, self.q, 1)
             elif isinstance(other, Rational):
-                #TODO: this can probably be optimized more
-                return Rational(self.p*other.q + self.q*other.p, self.q*other.q)
+                ap, aq = self.p, self.q
+                bp, bq = other.p, other.q
+                g = math.gcd(aq, bq)
+                if g == 1:
+                    p = ap*bq + aq*bp
+                    q = bq*aq
+                    return Rational(p, q)
```

Comment:
In all cases, since gcd has been removed, use `Rational(p, q, gcd=1)` so `Rational` does not have to recalculate the `gcd`.

Reply:
https://github.com/sympy/sympy/issues/12602#issuecomment-1874364587 According to this comment is it meaningful these changes.

Reply:
I am not following super closely, but this PR seems to have misplaced logic. Rational will handle gcd extraction so if there is something that needs to be optimized wrt to how gcd is calculated, the gcd called in Rational is where the the  change is needed.